wagtail/admin/views/pages/delete.py

```diff
@@ -17,32 +18,67 @@ def delete(request, page_id):
         raise PermissionDenied
 
     with transaction.atomic():
-        for fn in hooks.get_hooks('before_delete_page'):
+        for fn in hooks.get_hooks("before_delete_page"):
             result = fn(request, page)
-            if hasattr(result, 'status_code'):
+            if hasattr(result, "status_code"):
                 return result
 
         next_url = get_valid_next_url_from_request(request)
 
-        if request.method == 'POST':
+        pages_to_delete = {page}
+
+        # The `construct_synced_page_tree_list` hook returns translation and
+        # alias pages when the action is set to "delete"
+        if getattr(settings, 'WAGTAIL_I18N_ENABLED', False):
+            for fn in hooks.get_hooks("construct_synced_page_tree_list"):
+                fn_pages = fn([page], "delete")
+                if fn_pages and isinstance(fn_pages, dict):
+                    for additional_pages in fn_pages.values():
+                        pages_to_delete.update(additional_pages)
+
+        pages_to_delete = list(pages_to_delete)
+
+        if request.method == "POST":
             parent_id = page.get_parent().id
+            # Delete the source page.
             action = DeletePageAction(page, user=request.user)
             # Permission checks are done above, so skip them in execute.
             action.execute(skip_permission_checks=True)
 
-            messages.success(request, _("Page '{0}' deleted.").format(page.get_admin_display_title()))
+            # Delete translation and alias pages if they have the same parent page.
+            if getattr(settings, 'WAGTAIL_I18N_ENABLED', False):
+                parent_page_translations = page.get_parent().get_translations()
```

Comment:
I still feel this should move to the simple_translation hook implemnentation.

tl;dr the delete view becomes a simple actor on the list of pages that need action. Should a developer decide to provide more, then that's ok